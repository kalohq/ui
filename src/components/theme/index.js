/* @flow */
export {default} from './theme';
