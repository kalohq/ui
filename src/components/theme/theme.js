/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
