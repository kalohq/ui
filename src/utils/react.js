import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
