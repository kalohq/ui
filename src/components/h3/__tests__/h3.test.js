/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import H3 from 'components/h3';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('H3', () => {
  const defaultProps = {};
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <H3 {...defaultProps} {...props}>
            An H3 Tag
          </H3>
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should allow weight to be overridden', () => {
    const element = create({
      weight: 'light',
    });
    expect(element).toMatchSnapshot();
  });
});
