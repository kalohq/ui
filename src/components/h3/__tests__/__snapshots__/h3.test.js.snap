// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H3 should allow weight to be overridden 1`] = `
.emotion-2 {
  display: block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 300;
  font-size: 14px;
  color: rgb(55,69,97);
  line-height: 1.25em !important;
}

.emotion-2:hover {
  text-decoration: inherit;
}

.emotion-2 strong {
  font-weight: 500;
}

.emotion-2 > i {
  position: relative;
  left: 2px;
  vertical-align: top;
}

<h3
  className="emotion-0 emotion-1 emotion-2"
  color="navy700"
  onClick={undefined}
  size="small"
  style={Object {}}
>
  An H3 Tag
</h3>
`;

exports[`H3 should render shallow component ok 1`] = `
.emotion-2 {
  display: block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 600;
  font-size: 14px;
  color: rgb(55,69,97);
  line-height: 1.25em !important;
}

.emotion-2:hover {
  text-decoration: inherit;
}

.emotion-2 strong {
  font-weight: 500;
}

.emotion-2 > i {
  position: relative;
  left: 2px;
  vertical-align: top;
}

<h3
  className="emotion-0 emotion-1 emotion-2"
  color="navy700"
  onClick={undefined}
  size="small"
  style={Object {}}
>
  An H3 Tag
</h3>
`;
